pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -2413,11 +2413,16 @@ private void recoverProcessOpSendMsgFrom(ClientCnx cnx, MessageImpl from, long e
     private boolean isMessageSizeExceeded(OpSendMsg op) {
         if (op.msg != null && !conf.isChunkingEnabled()) {
             int messageSize = op.getMessageHeaderAndPayloadSize();
-            if (messageSize > getMaxMessageSize()) {
+            int maxMessageSize = getMaxMessageSize();
+            if (messageSize > maxMessageSize) {
+                // Log the message size exceeding the limit
+                log.warn("Message size exceeded: Producer {} attempted to send a message of {} bytes, "
+                                + "exceeding the configured maximum size of {} bytes on topic {}.",
+                        producerName, messageSize, maxMessageSize, topic);
```

Comment:
Seems useless for this log, the client could get the exception with log below.

Reply:
+1, the exception can be propagated to the send oeration.